Foundation/URLSession/TransferState.swift

```diff
@@ -100,6 +108,49 @@ extension _HTTPURLProtocol._TransferState {
     }
 }
 
+// specific to FTP
+extension _FTPURLProtocol._TransferState {
+
+    enum FTPHeaderCode: Int {
+        case transferCompleted = 226
+        case openDataConnection = 150
+        case fileStatus = 213
+        case syntaxError = 500// 500 series FTP Syntax errors
+        case errorOccurred = 400 // 400 Series FTP transfer errors
+    }
+
+    /// Appends a header line
+    ///
+    /// Will set the complete response once the header is complete, i.e. the
+    /// return value's `isHeaderComplete` will then by `true`.
+    ///
+    /// - Throws: When a parsing error occurs
+    func byAppendingFTP(headerLine data: Data, expectedContentLength: Int64) throws -> _NativeProtocol._TransferState {
+        let line = String(data: data, encoding: String.Encoding.utf8)
+        //FTP Status code 226 marks the end of the transfer
+        if (line?.starts(with: String(FTPHeaderCode.transferCompleted.rawValue)))! {
+            return self
+        }
+        //FTP Status code 213 marks the end of the header and start of the transfer on data port
+        func  isCompleteHeader(_ headerLine: String) ->Bool {
```

Comment:
Nit: please fix the spacing here; operators like `->` need one space on either side--I'm not certain that Swift currently complains about `->` specifically, but it certain does about other infix operators

While you're at it, please remove the extra space after `func` and after `return` on the following line.